core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -840,6 +841,40 @@ public static void resetCountersAndGauges() {
     }
   }
 
+  public static class MultiTimerContext implements AutoCloseable {
```

Comment:
As the name of this class, would you like to support [0, n] timers?

```java
    private final Timer[] mTimers;
    public MultiTimerContext(Timer... timers) {
      for (Timer timer : timers) {
        Preconditions.checkNotNull(timer, "timer");
      }
      mTimers = timers;
      mStartTime = System.nanoTime();
    }

    public long stop() {
      final long elapsed = System.nanoTime() - mStartTime;
      for (Timer timer : mTimers) {
        timer.update(elapsed, TimeUnit.NANOSECONDS);
      }
      return elapsed;
    }
```

Reply:
good call